app/models/exercise.rb

```diff
@@ -35,6 +35,7 @@ class Exercise < ApplicationRecord
   has_many :pair_programming_waiting_users
 
   scope :with_submissions, -> { where('id IN (SELECT exercise_id FROM submissions)') }
+  scope :with_programming_groups, -> { where('id IN (SELECT DISTINCT exercise_id FROM programming_groups)') }
```

Comment:
I would argue that the `DISTINCT` operator here is not necessary and rather adds unrequired overhead. In this case, I would suggest dropping it.